packages/react/hover-card/src/HoverCard.tsx

```diff
@@ -0,0 +1,229 @@
+import * as React from 'react';
+import { composeEventHandlers } from '@radix-ui/primitive';
+import { createContext } from '@radix-ui/react-context';
+import { useControllableState } from '@radix-ui/react-use-controllable-state';
+import * as PopperPrimitive from '@radix-ui/react-popper';
+import { Portal } from '@radix-ui/react-portal';
+import { Presence } from '@radix-ui/react-presence';
+import { Primitive, extendPrimitive } from '@radix-ui/react-primitive';
+import type * as Polymorphic from '@radix-ui/react-polymorphic';
+
+/* -------------------------------------------------------------------------------------------------
+ * HoverCard
+ * -----------------------------------------------------------------------------------------------*/
+
+const HOVERCARD_NAME = 'HoverCard';
+
+type HoverCardContextValue = {
+  open: boolean;
+  onOpenChange(open: boolean): void;
+  onMouseEnter(): void;
+  onMouseLeave(): void;
+};
+
+const [HoverCardProvider, useHoverCardContext] = createContext<HoverCardContextValue>(
+  HOVERCARD_NAME
+);
+
+type HoverCardOwnProps = {
+  open?: boolean;
+  defaultOpen?: boolean;
+  onOpenChange?: (open: boolean) => void;
+  openDelay?: number;
+  closeDelay?: number;
+};
+
+const HoverCard: React.FC<HoverCardOwnProps> = (props) => {
+  const {
+    children,
+    open: openProp,
+    defaultOpen,
+    onOpenChange,
+    openDelay = 700,
+    closeDelay = 300,
+  } = props;
+  const enterInvokedRef = React.useRef(false);
+  const enterTimerRef = React.useRef(0);
+  const leaveTimerRef = React.useRef(0);
+
+  const [open = false, setOpen] = useControllableState({
+    prop: openProp,
+    defaultProp: defaultOpen,
+    onChange: onOpenChange,
+  });
+
+  const handleMouseEnter = React.useCallback(() => {
+    clearTimeout(leaveTimerRef.current);
+
+    enterTimerRef.current = window.setTimeout(() => {
+      setOpen(true);
+      enterInvokedRef.current = true;
+    }, openDelay);
+  }, [openDelay, setOpen]);
+
+  const handelMouseLeave = React.useCallback(() => {
+    clearTimeout(enterTimerRef.current);
+
+    leaveTimerRef.current = window.setTimeout(() => {
+      if (enterInvokedRef.current) {
+        setOpen(false);
+        enterInvokedRef.current = false;
+      }
+    }, closeDelay);
+  }, [closeDelay, setOpen]);
+
+  // cleanup any queued state updates on unmount
+  React.useEffect(() => {
+    return () => {
+      clearTimeout(enterTimerRef.current);
+      clearTimeout(leaveTimerRef.current);
+    };
+  }, []);
+
+  return (
+    <PopperPrimitive.Root>
+      <HoverCardProvider
+        open={open}
+        onOpenChange={setOpen}
+        onMouseEnter={handleMouseEnter}
+        onMouseLeave={handelMouseLeave}
+      >
+        {children}
+      </HoverCardProvider>
+    </PopperPrimitive.Root>
```

Comment:
One thing worth mentioning for future reference. We used to put our provider inside like this so that it was only the children receiving the context, however, we since introduced a `Slot` component. For `Slot` to work, the component that wraps the `children` has to be the primitive part that the consumer can apply `as` to. 

In this case, it doesn't really matter because the `PopperPrimitive.Root` only renders a provider and not a node but this is good knowledge to have. Might be easier to go into more detail on a call on this one though if you're interested 🙂

Reply:
Interestingly I almost raised this on your PRs yesterday Jenna, but then I realised that they were root components which didn't render a node. I wonder if it's worth doing it on these too just for consistency as it caught my attention wrongly (especially when I saw `{children}`)

Reply:
@jjenzz explained this well to me earlier but just so I'm totally clear, in this example I should switch the order of `PopperPrimitive.Root` with `HoverCardProvider` to maintain consistency?

Reply:
Yeah, let's do that @andy-hook and we can do the same in our other components in another PR later to save confusion in future PRs.